examples/tensorflow/summarization/run_summarization.py

```diff
@@ -0,0 +1,682 @@
+    train_signature = {
+        feature: tf.TensorSpec(shape=(None,), dtype=tf.int32)
+        for feature in dataset.features
+        if feature != "special_tokens_mask"
+    }
+    if (
+        model is not None
+        and "decoder_input_ids" not in train_signature
+        and hasattr(model, "prepare_decoder_input_ids_from_labels")
+    ):
+        train_signature["decoder_input_ids"] = train_signature["labels"]
+    # This may need to be changed depending on your particular model or tokenizer!
+    padding_values = {
+        key: tf.convert_to_tensor(tokenizer.pad_token_id if tokenizer.pad_token_id is not None else 0, dtype=tf.int32)
+        for key in train_signature.keys()
+    }
+    padding_values["labels"] = tf.convert_to_tensor(-100, dtype=tf.int32)
+    train_signature["labels"] = train_signature["input_ids"]
+    train_signature = (train_signature, train_signature["labels"])
+    options = tf.data.Options()
+    options.experimental_distribute.auto_shard_policy = tf.data.experimental.AutoShardPolicy.OFF
+    tf_dataset = (
+        tf.data.Dataset.from_generator(train_generator, output_signature=train_signature)
+        .with_options(options)
+        .padded_batch(
+            batch_size=total_batch_size,
+            drop_remainder=True,
+            padding_values=(padding_values, np.array(-100, dtype=np.int32)),
+        )
+        .repeat(int(num_epochs))
+    )
+    return tf_dataset
+
+
+# endregion
+
+
+def main():
+    # region Argument parsing
+    # See all possible arguments in src/transformers/training_args.py
+    # or by passing the --help flag to this script.
+    # We now keep distinct sets of args, for a cleaner separation of concerns.
+
+    parser = HfArgumentParser((ModelArguments, DataTrainingArguments, Seq2SeqTrainingArguments))
+    if len(sys.argv) == 2 and sys.argv[1].endswith(".json"):
+        # If we pass only one argument to the script and it's the path to a json file,
+        # let's parse it to get our arguments.
+        model_args, data_args, training_args = parser.parse_json_file(json_file=os.path.abspath(sys.argv[1]))
+    else:
+        model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+    # endregion
+
+    # region Logging
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+    logger.setLevel(logging.INFO)
+    datasets.utils.logging.set_verbosity(logging.INFO)
+    transformers.utils.logging.set_verbosity(logging.INFO)
+
+    # Log on each process the small summary:
+    logger.info(f"Training/evaluation parameters {training_args}")
+    # endregion
+
+    # region T5 special-casing
+    if data_args.source_prefix is None and model_args.model_name_or_path in [
+        "t5-small",
+        "t5-base",
+        "t5-large",
+        "t5-3b",
+        "t5-11b",
+    ]:
```

Comment:
This isn't very robust, as any community models wouldn't work with that. If you're looking for all T5 models and you want to check early on if this is set, you could check the configuration as soon as it's instantiated and look for the `model_type`:

```py
>>> config.model_type
't5'
```

Reply:
This test was heavily debated on the PyTorch side, so cc-ing @patil-suraj and @patrickvonplaten

Reply:
I've implemented all the other changes locally, but I'll hold off on pushing them until we resolve this one!

Reply:
I would avoid this check here. prefixes are only really required for the t5V1 models. The newer T5V1.1 does not use prefixes.  As Lysandre said above check is not robust and we can't also check `model_type==t5` as it'll catch all T5 models both v1 and V1.1.

So IMO, better to just leave it to the user with a note in the readme.

Reply:
Will do!

Reply:
Why is the test in the PyTorch example then?

Reply:
Aah, I did not realize that we still have that in the PyTorch example :(

Reply:
I think we should leave this as is here, since there was a very heated debate on the PyTorch side. If we want to remove it from all examples together, it can be done in a separate PR.

Reply:
Fine by me then. Let's not debate on this anymore :)

Reply:
Will un-do!